### PR TITLE
Remove selected elements and page selection as separate history steps

### LINKED ISSUE #67: Correct history management
## Feature description

The history currently remembers too many states compared to the UX. We don't want to remember every change in selection nor every page change, but when undoing, the selection and page change associated with the currently "undone" state should be recreated.

Please see UX for details.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* Selecting an element or changing page should not count as an undo state in itself in the history.
* Selection and current page does however need to be part of every state in the undo history.

## Implementation brief

* It can probably be done simply by remembering current page and selection in refs, but only actually updating the history state on changes to the `pages` array.

Something like this might work ([compare with existing](https://github.com/ampproject/amp-wp/blob/develop-stories/assets/src/edit-story/app/story/effects/useHistoryEntry.js)):

```js
function useHistoryEntry( {
	currentPageIndex,
	pages,
	selectedElementIds,
} ) {
	const { actions: { appendToHistory } } = useHistory();
	const currentPageIndexRef = useRef();
	const selectedElementIdsRef = useRef();
	useEffect( () => {
		currentPageIndexRef.current = currentPageIndex;
		selectedElementIdsRef.current = selectedElementIds;
	}, [ currentPageIndex, selectedElementIds ] );
	useEffect( () => {
		appendToHistory( {
			currentPageIndex,
			pages,
			selectedElementIds,
		} );
	}, [ appendToHistory, currentPageIndex, pages, selectedElementIds ] );
	useEffect( () => {
		appendToHistory( {
			currentPageIndex: currentPageIndexRef.current,
			selectedElementIds: selectedElementIdsRef.current,
			pages,
		} );
	}, [ appendToHistory, pages ] );
}
```

Then only changes to `pages` will actually create a new entry in the history, but any changes to current page or selection will be remembered and persisted along with those major page changes.

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

### PR DESCRIPTION
**Pull request by miina**
Fixes #67

### CODE REVIEW

File: assets/src/edit-story/app/story/effects/useHistoryEntry.js
Comment (miina) [inline assets/src/edit-story/app/story/effects/useHistoryEntry.js:48]: This should work in theory, however, `pages` seems to be changing with selection as well, causing the `useEffect` (and `appendToHistory`) to trigger.

Comment (miina) [reply assets/src/edit-story/app/story/effects/useHistoryEntry.js:48]: @dvoytenko FI: It looks like your proposed PR of #621 fixes the issue of `pages` incorrectly being changed/re-initialized which blocks this PR. (At least that's how it looked like when testing)

It probably makes sense to wait until #621 gets merged to merge this one.

Comment (dvoytenko) [reply assets/src/edit-story/app/story/effects/useHistoryEntry.js:48]: @miina could you pls clarify whey the pages are incorrectly changed now? Otherwise, certainly ok to block until #621 is merged. I'll try to get it in sooner.

Comment (miina) [reply assets/src/edit-story/app/story/effects/useHistoryEntry.js:48]: > @miina could you pls clarify whey the pages are incorrectly changed now?

@dvoytenko Hmm, odd, actually, it looks like there is no issue with `pages` anymore when testing.
Previously, when testing, creating a new history was triggered even if nothing was actually changed on a Page. Testing showed that `pages` was the variable triggering it. I did not investigate further since it seemed to work OK on #621, however, it seems to work as expected currently as well.

Will mark this ready for review then.

File: assets/src/edit-story/components/library/dropzone/index.js
Comment (miina) [inline assets/src/edit-story/components/library/dropzone/index.js:1]: Note: this was just an empty file.

File: assets/src/edit-story/app/story/effects/useHistoryEntry.js
Comment (dvoytenko) [reply assets/src/edit-story/app/story/effects/useHistoryEntry.js:48]: Ok. This all looks pretty good then. But why are the capabilities here? Is there an issue already open to move them elsewhere?

**Review by dvoytenko — CHANGES_REQUESTED** [review]

**Comment by miina** [comment]
> But why are the capabilities here? Is there an issue already open to move them elsewhere?

Yes, there is another issue, see #196 (also a closed duplicate #630) and [previous discussion](https://github.com/ampproject/amp-wp/pull/4122#discussion_r367949420).

**Review by dvoytenko — APPROVED** [review]

**Comment by dvoytenko** [comment]
@miina LGTM. But we also need to fix #196 asap then. Let's bring this up at the next fixit/sprint.

**Pull request merged**
